crates/store/src/state.rs

```diff
@@ -96,17 +96,113 @@ pub struct TransactionInputs {
     pub found_unauthenticated_notes: BTreeSet<NoteId>,
 }
 
+/// Blockchain wrapper for `Mmr` .c
+#[derive(Debug, Clone)]
+pub struct Blockchain(Mmr);
+
+impl Blockchain {
+    /// Returns a new Blockchain.
+    pub fn new(chain_mmr: Mmr) -> Self {
+        Self(chain_mmr)
+    }
+
+    /// Returns the tip of the chain, i.e. the number of the latest block in the chain.
+    pub fn chain_tip(&self) -> BlockNumber {
+        let block_number: u32 = (self.0.forest() - 1)
+            .try_into()
+            .expect("chain_mmr always has, at least, the genesis block");
+
+        block_number.into()
+    }
+
+    /// Returns the chain length.
+    pub fn chain_length(&self) -> BlockNumber {
+        let latest = self.chain_tip();
+        latest + 1
```

Comment:
I believe `BlockNumber::child()` exists. 

```suggestion
        self.chain_tip().child()
```

Though I'm unsure if we should have both `chain_tip` and `chain_length`? I guarantee someone's going to confuse the two. 

----------

--edit-- in fact further down we have `let chain_length = state.blockchain.chain_tip() as usize;`